client-api/src/main/java/org/sonarsource/sonarlint/core/client/api/util/FileUtils.java

```diff
@@ -39,13 +42,54 @@ private FileUtils() {
     // utility class, forbidden constructor
   }
 
-  public static void moveDir(Path temp, Path dest) {
+  public static void moveDir(Path src, Path dest) {
     try {
-      Files.move(temp, dest, StandardCopyOption.ATOMIC_MOVE);
-    } catch (AtomicMoveNotSupportedException e) {
-      throw new IllegalStateException("Atomic move not supported for " + temp + " to " + dest, e);
+      try {
+        Files.move(src, dest, StandardCopyOption.ATOMIC_MOVE);
+      } catch (AtomicMoveNotSupportedException e) {
+        // Fallback to non atomic move
+        Files.walkFileTree(src, new MoveRecursivelyVisitor(src, dest));
+      }
     } catch (IOException e) {
-      throw new IllegalStateException("Unable to move " + temp + " to " + dest, e);
+      throw new IllegalStateException("Unable to move " + src + " to " + dest, e);
+    }
+  }
+
+  private static class MoveRecursivelyVisitor extends SimpleFileVisitor<Path> {
+    private final Path fromPath;
+    private final Path toPath;
+    private final CopyOption[] copyOptions;
+
+    public MoveRecursivelyVisitor(Path fromPath, Path toPath, CopyOption... copyOptions) {
+      this.fromPath = fromPath;
+      this.toPath = toPath;
+      this.copyOptions = copyOptions;
+    }
+
+    @Override
+    public FileVisitResult preVisitDirectory(Path dir, BasicFileAttributes attrs) throws IOException {
+      Path targetPath = toPath.resolve(fromPath.relativize(dir));
+      if (!Files.exists(targetPath)) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Replace this with a call to the "toFile().exists()" method [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS3725)